### DISCUSSION TITLE
Best Practices : Akash Governance Proposal's

### DISCUSSION THREAD

**Discussion by CaptainSplince** [akash-network/community#170] [Governance Proposals]
_Updated October 2024_

# Overview

This document gives best practices for how to prepare for requesting resources from the Akash community pool at present. Below are some recommended guidelines to help structure your ideas into governance proposals. 

# Research and Ideation

We know you're excited, but before you start drafting your proposal, it's important you research what tools, support and resources are available to you. Below are considerations and questions that you need to prepare for before you create your proposal for community review.

## Prepare the prerequisites

- Join [Akash's Discord.](https://discord.akash.network/) By joining our discord, you'll have access to a wealth of resources and the opportunity to engage our community with questions and discuss your early-stage ideas. This is a good place to gut-check your ideas.
- Familiarize yourself with the current [Akash community groups](https://akash.network/development/welcome/#community-groups) and [GH discussions.](https://github.com/orgs/akash-network/discussions)
- Do your research and answer these key questions:
  - What is the problem you're trying to solve?
  - How will the work benefit the Akash Network and / or Akash Community?
  - Play devil's advocate: are there reasons your idea might not work?
  - Is work already being done somewhere in the Akash Community?
    - If you're not sure ask in the [Ideas channel in Github](https://github.com/orgs/akash-network/discussions/categories/ideas) 
  - Is the project you're working on open-source or closed?
    - Open Source: Projects should be built in the spirit of open-sourcing your work and ideas for the community to build off of.
    - Closed Source: Projects that do not allow community members access your code
  - Public Good: There should be a clear connection between your proposal and the value it generates for the broader community.
- Review previous proposals that have passed or been presented on [Github](https://github.com/orgs/akash-network/discussions/categories/governance-proposals)
- Develop projected outcomes that are specific, measurable, attainable, relevant and time-bound 
  - Projected timelines for deliverables that are broken into quarters are best. 
  - For example: If you think your project could take a year to complete, break it down into discrete quarterly milestones that can be measured and tracked.
- Is your proposal a project or a program?
  - A project has a timeline and an end date. 
  - A program is an ongoing initiative with no specific end date in mind. 
  - For programs: clearly identify metrics you can track against that will allow the community determine success or not. to demonstrate the ongoing success of your program. 
  - Funding for programs are typically requested quarterly , so be prepared to submit on-chain proposals quarterly for community pool spend referencing past metrics, etc.
- Join the [Akash Discord](https://discord.akash.network/), GH ["Discussions"](https://github.com/orgs/akash-network/discussions), and/or WG/SIG meetings for detailed conversations that should give feedback into the draft proposal.

## Join Special Interest Group (SIG) or Work Group (WG) channel

- Join one of the SIGs or WG channels and introduce yourself and your idea, gathering feedback to iterate and refine.
- [Full list of SIGs and WGs](https://akash.network/community/) (under "Core Community Groups")
- Join SIG or WG monthly call
  - This is an opportunity to get feedback for your early idea directly from the community
  - Events happen monthly.

# Proposal Creation Phase

Once you have done all your research on the who, what and where of Akash governance proposals, it's time to draft your proposal!

- Your proposal should be a collection of interactions you've discussed with the community, or core teams in the SIG/WG's you've attended leading up to this point. 
- The Akash SIGs/WG's and Steering Committee are resources, use them for guidance.
- Create an initial draft of the proposal 
  - Try to include a TL:DR
  - Timeline
  - Amount
  - Goal/Mission
  - If it is a technical solution (feature request, enhancement, etc), detail out the deliverables:
    - Overview of the project
    - Create a document known as a PRD - Product Requirement Document (if needed), to describe the proposed solution in more technical detail.
  - Include estimated hours, hourly rate, and other costs for a total value to show how you arrived at the funding request. 
    - While there are many ways to demonstrate estimated costs, the clearer and more detailed you are, the easier it is for validators and AKT holders to understand your ask
- Include supporting references for the Github discussion:
  - Feel free to attach video and notes from SIG/WG meetings you've attended
  - Slide decks 
  - GH discussions
  - Demos or proof of concepts
- Finalize your funding request
  - Total funding should be broken down into components, whether that is hours, deliverables, milestones, etc.
  - While there are many ways to demonstrate estimated costs, the clearer and more detailed you are, the easier it is for the Akash community to understand your ask
  - Will you need multiple payments or one lump sum payment to complete your project?
    - NOTE: If multiple payments are required, you will need to submit additional governance proposals
    - NOTE: There is a 7-day delay from submission of a proposal to the end of the voting period. Regardless of price fluctuation, AKT amount released is the same as the requested amount 
- Responsible Liquidation
  - If you are awarded public funds, and need to liquidate to a stable currency, please do so in a responsible manner.
  - If you don't know how, please ask.
- Returning funds
  - If your proposal says you will, you should. Everything is on-chain and reputation is a social forcing function
  - One reason for returning funds is due to failed implementation, milestones not being hit, deliverables not received, etc.
  - Another valid reason is there were leftover funds than what the proposal required. This could happen when a certain USD/Stable amount is desired, reached, and there are leftover AKT.
  - CLI command - akash tx distribution fund-community-pool xxxxxuakt --from=akash1234….
- Put the finishing touches on your proposal
  - Post your proposal to Github under the [Governance Proposals category.](https://github.com/orgs/akash-network/discussions/categories/governance-proposals)
  - Be sure you include a timeline of when you want to get your proposal on chain for a vote so the community has enough time to review and provide you with feedback. 

# Review Period

Now that you've drafted your proposal, it's time to get feedback from the community. All governance proposals are posted to [Github Discussions](https://github.com/orgs/akash-network/discussions) for review. 

- Connect with the Akash community to get comments on your Github proposal discussion
  - Join [SIGs and WGs](https://akash.network/development/welcome/#community-groups) calls and present your idea
  - Post a link to your discussion to Discord, Telegram, X or other relevant community channels and encourage feedback directly on Github
  - NOTE: Proposals that do not receive community feedback are less likely to pass due to lack of understanding or interest. (Market your idea!)
- Respond to any questions and comments from the Akash Community in your Github discussion  
- Incorporate feedback into your proposal
  - We suggest posting any edits in the original Github discussion OR as comments, as long as you reference the edits made as a response to feedback.
- Briefly talk about your proposal at a monthly [Akash Steering Committee meeting](https://github.com/akash-network/community/tree/main/committee-steering) to secure support from the Akash Steering Committee. While not required, support from the Steering Committee is highly encouraged.

# Submitting on chain

Now that you've received community feedback and adjusted your proposal accordingly, it's time to get your proposal on chain for a vote.

## How to submit a proposal
- Currently there are two ways to draft a Community Pool Spend governance proposal for Akash Network
  - Using CLI
  - Minstcan.io "Create Proposal" feature for Community Pool Spend
- If you have questions on either of these, reach out to the community
- Create the proposal yourself. If you don't know how to do this, reach out to the general channel on Discord for help. 
- Prerequisite: all proposals require a 1000 AKT deposit to go on-chain for a vote
  - If the vote passes - you get your AKT back
  - If the vote fails - you get your AKT back
  - If the vote is no with veto - you lose your AKT (this is rare and usually happens due to malicious or scam proposals) 
- Once your proposal is on chain, let the community know on Discord, Telegram, and Twitter. Overclock's team, and Akash Network Insiders are proactive in supporting proposals designed to benefit the community and network growth.
- Be sure to link to your proposal's github discussion in the description of your on chain proposal

# Proposal status and reporting

Congratulations! If your proposal passed, you now have funding to support your efforts. Now the fun begins! However, it's important to share with the community how your projects are coming along. Below are some notes on what you should be doing as your project progresses

## If your proposal passes
- If you receive Community Pool funding you should provide some kind of reporting on status, funding, etc.
- AKT or other funding reporting should go here [Community Pool Spend Reporting](https://github.com/akash-network/community/tree/main/community-pool-spend-reporting) in Github.
- Projects updates should go in Github and Discord and should be shared live with the community during SIG and WG calls

## If proposal fails
- Take community feedback and try again. 

# Project completion

You put in the time and effort creating a fantastic project. Kudos! Now it's time to share your results. Below are a few things you can to do spread the word!

- Draft a case study
- Draft a blog post
- Record a demo video
- Share your finished project in a SIG/WG call
- Add documentation to the [Akash Network docs](https://akash.network/docs/) page

Once your project is complete, the fun doesn't have to stop there. Spend some time thinking about other areas that you could improve Akash. You can use the above process for more proposals. 

Thanks for reading and good luck!

**Comment by wastrelwander1** [comment]
This guide has been so helpful. Thank you for putting it together. You mention that funding from the Community Pool is typically released quarterly. Is this usually at the end of each calendar year quarter, e.g. end of Sept, end of Dec, etc?

**Reply by piber-dev** [reply]
hi @wastrelwander1 it would be best to ask questions like this on the discord - which i believe you already did and got a response for. not all threads are viewed regularly. funding from the community pool is not typically released quarterly, funding is released as soon as the 7-day voting period is up and the proposal passed. however, many recurring programs request funds quarterly for reasons such as to not request a lot of funds at one point and then not being able to deliver on the expected results.

**Reply by wastrelwander1** [reply]
Ok brilliant, thank you so much piber!
